STACpopulator/stac_utils.py

```diff
@@ -205,6 +206,18 @@ class MySTACItem(Item):
 
         self.item = item
 
+    def item_link(self) -> pystac.Link:
+        """Return STAC Link to the catalog entry for the dataset."""
+        url = self.attrs["catalog_url"]
+        parts = url.split("/")
+        service = parts[parts.index("catalog") - 1]
+
+        link = pystac.Link(rel="source",
+                           target=url,
+                           media_type="text/html",
```

Comment:
In the case of the STAC Item that represents a NetCDF, `source` should probably refer to `application/x-netcdf` with an URL of the actual file? This URL should be indicated at the top level of the NCML file as `@location` using `Dataset.to_cf_dict`.

If this change is done, instead of looking for `/catalog/` in the URL, the `/dodsC/` must be considered.

This code can be reused though, to inject `source` in the STAC Collection, which would be a XML THREDDS catalog URL.

Reply:
I also found that the NCML has a `id` attribute in `THREDDSMetadata` with value `birdhouse/testdata/xclim/cmip6/sic_SImon_CCCma-CanESM5_ssp245_r13i1p2f1_2020.nc`. 
This seems more reliable than parsing the URL ourselves.

Reply:
1. The href is not the URL of the actual netCDF file, but of its catalog entry, so not sure the mime should be application/netcdf. The netcdf link is listed among the Assets.
2.  Yes, I'm using that ID already.

Reply:
When loading in `xncml.Dataset`, the `"@location"` in returned attributes directly at the root is the NetCDF file URL. 

This is the same `location` attribute directly at the top of: 
https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/ncml/birdhouse/testdata/xclim/cmip6/sic_SImon_CCCma-CanESM5_ssp245_r13i1p2f1_2020.nc?catalog=https%3A%2F%2Fpavics.ouranos.ca%2Ftwitcher%2Fows%2Fproxy%2Fthredds%2Fcatalog%2Fbirdhouse%2Ftestdata%2Fxclim%2Fcmip6%2Fcatalog.html&dataset=birdhouse%2Ftestdata%2Fxclim%2Fcmip6%2Fsic_SImon_CCCma-CanESM5_ssp245_r13i1p2f1_2020.nc

The link actually returned uses `dodsC` service, so the response from THREDDS (https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/fileServer/birdhouse/testdata/xclim/cmip6/sic_SImon_CCCma-CanESM5_ssp245_r13i1p2f1_2020.nc) that I get is in fact a 400 HTML 😞 

What we could do is employ the `fileServer` endpoint instead, where that would be the real `source` in `application/x-netcdf` (https://pavics.ouranos.ca/twitcher/ows/proxy/thredds/fileServer/birdhouse/testdata/xclim/cmip6/sic_SImon_CCCma-CanESM5_ssp245_r13i1p2f1_2020.nc).

I think the NCML XML href and the THREDDS Catalog HTML href would be relevant in `links` as `rel=describedby` and `rel=alternate` respectively. The NCML href could also use `convertedfrom` actually!

The `rel=collection` href could also be inserted since you have it.

Reply:
https://github.com/crim-ca/ncml2stac/blob/main/notebooks/ncml2stac.ipynb contains the updated `item_link` implementation using `@location` attribute